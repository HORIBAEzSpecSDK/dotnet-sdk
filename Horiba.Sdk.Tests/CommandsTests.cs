namespace Horiba.Sdk.Tests;

public class CommandsTests
{
    [Fact]
    public async Task GivenIclInfoCommand_WhenSerializing_ThenCommandIdAndNameAreCorrect()
    {
        // Arrange
        var iclInfoCommand = new IclInfoCommand();
        
        // Act
        var jsonCommand = iclInfoCommand.ToJson();
        
        // Assert
        jsonCommand.MatchSnapshot();
    }
    
    [Fact]
    public async Task GivenDeviceInfoCommand_WhenSendingUsingCommunicator_ThenReceivesExpectedResponse()
    {
        // Arrange
        var communicator = new WebSocketCommunicator();
        await communicator.OpenConnectionAsync();
        var command = new IclInfoCommand();

        // Act
        var response = await communicator.SendAsync(command);

        // Assert
        response.MatchSnapshot();
    }
}