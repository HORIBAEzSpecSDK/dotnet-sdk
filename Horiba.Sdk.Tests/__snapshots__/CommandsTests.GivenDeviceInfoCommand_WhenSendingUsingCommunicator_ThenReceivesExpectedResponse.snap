{
  "id": 1,
  "command": "icl_info",
  "results": {
    "nodeAlias": "ICL",
    "nodeApiVersion": 300,
    "nodeBuilt": "Mar  1 2024-09:33:43",
    "nodeDescription": "Instrument Control Library",
    "nodeId": 1,
    "nodeVersion": "2.0.0.119.46581d06"
  },
  "errors": []
}
